Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -371,6 +380,9 @@ void RenderBox::styleDidChange(StyleDifference diff, const RenderStyle* oldStyle
     if (isDocElementRenderer || isBodyRenderer) {
 #if ENABLE(DARK_MODE_CSS)
         view().frameView().recalculateBaseBackgroundColor();
+
+        if (oldStyle && oldStyle->colorScheme() != newStyle.colorScheme())
+            view().frameView().invalidateScrollCorner(view().frameView().scrollCornerRect());
```

Comment:
If you take my suggestion in https://github.com/WebKit/WebKit/pull/15264/files#r1268696498, I expect this code can be removed, as the `view().frameView().recalculateScrollbarOverlayStyle()` should take care of this.